Model/Indexer/Invalidate.php

```diff
@@ -0,0 +1,177 @@
+use Nosto\Tagging\Model\ResourceModel\Magento\Product\CollectionFactory as ProductCollectionFactory;
+use Nosto\Tagging\Model\Service\Index as NostoServiceIndex;
+use Nosto\Tagging\Model\Service\IndexerStatusServiceInterface;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * Class Invalidate
+ * This class is responsible for listening to product changes
+ * and setting the `is_dirty` value in `nosto_product_index` table
+ * @package Nosto\Tagging\Model\Indexer
+ */
+class Invalidate extends AbstractIndexer
+{
+    const INDEXER_ID = 'nosto_index_product_invalidate';
+
+    /** @var NostoServiceIndex */
+    private $nostoServiceIndex;
+
+    /** @var ProductCollectionFactory */
+    private $productCollectionFactory;
+
+    /** @var NostoHelperAccount */
+    private $nostoHelperAccount;
+
+    /** @var InvalidateModeSwitcher */
+    private $modeSwitcher;
+
+    /**
+     * Invalidate constructor.
+     * @param NostoHelperAccount $nostoHelperAccount
+     * @param NostoHelperScope $nostoHelperScope
+     * @param NostoServiceIndex $nostoServiceIndex
+     * @param NostoLogger $logger
+     * @param ProductCollectionFactory $productCollectionFactory
+     * @param InvalidateModeSwitcher $modeSwitcher
+     * @param StoreDimensionProvider $dimensionProvider
+     * @param Emulation $storeEmulation
+     * @param ProcessManager $processManager
+     * @param InputInterface $input
+     * @param IndexerStatusServiceInterface $indexerStatusService
+     */
+    public function __construct(
+        NostoHelperAccount $nostoHelperAccount,
+        NostoHelperScope $nostoHelperScope,
+        NostoServiceIndex $nostoServiceIndex,
+        NostoLogger $logger,
+        ProductCollectionFactory $productCollectionFactory,
+        InvalidateModeSwitcher $modeSwitcher,
+        StoreDimensionProvider $dimensionProvider,
+        Emulation $storeEmulation,
+        ProcessManager $processManager,
+        InputInterface $input,
+        IndexerStatusServiceInterface $indexerStatusService
+    ) {
+        $this->nostoServiceIndex = $nostoServiceIndex;
+        $this->nostoHelperAccount = $nostoHelperAccount;
+        $this->productCollectionFactory = $productCollectionFactory;
+        $this->modeSwitcher = $modeSwitcher;
+        parent::__construct(
+            $nostoHelperAccount,
+            $nostoHelperScope,
+            $logger,
+            $dimensionProvider,
+            $storeEmulation,
+            $input,
+            $indexerStatusService,
+            $processManager
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getModeSwitcher(): ModeSwitcherInterface
+    {
+        return $this->modeSwitcher;
+    }
+
+    /**
+     * @inheritdoc
+     * @throws NostoException
+     */
+    public function doIndex(Store $store, array $ids = [])
+    {
+        if (!$this->nostoHelperAccount->nostoInstalledAndEnabled($store)) {
```

Comment:
At this point, isn't your `StoreDimensionProvider` only returning stores with accounts? This seems redundant.